parser/src/parser/tests/list_comprehension.rs

```diff
@@ -1,437 +1,234 @@
-                            Identifier("c".to_string()),
-                            AccessType::Default,
-                            0,
-                        ))),
-                    ),
-                    vec![
-                        (
-                            Identifier("i".to_string()),
-                            Iterable::Range(Range::new(0, 3)),
-                        ),
-                        (
-                            Identifier("c".to_string()),
-                            Iterable::Identifier(Identifier("c".to_string())),
-                        ),
-                    ],
-                )),
-            )),
-            BoundaryStmt::Constraint(BoundaryConstraint::new(
-                SymbolAccess::new(Identifier("a".to_string()), AccessType::Default, 0),
-                Boundary::First,
-                Add(
-                    Box::new(Add(
-                        Box::new(Add(
-                            Box::new(SymbolAccess(SymbolAccess::new(
-                                Identifier("x".to_string()),
-                                AccessType::Vector(0),
-                                0,
-                            ))),
-                            Box::new(SymbolAccess(SymbolAccess::new(
-                                Identifier("x".to_string()),
-                                AccessType::Vector(1),
-                                0,
-                            ))),
-                        )),
-                        Box::new(SymbolAccess(SymbolAccess::new(
-                            Identifier("x".to_string()),
-                            AccessType::Vector(2),
-                            0,
-                        ))),
-                    )),
-                    Box::new(SymbolAccess(SymbolAccess::new(
-                        Identifier("x".to_string()),
-                        AccessType::Vector(3),
-                        0,
-                    ))),
-                ),
-                None,
-            )),
-        ]),
-    ]);
-
-    ParseTest::new().expect_ast(source, expected);
+    let mut expected = Module::new(ModuleType::Root, SourceSpan::UNKNOWN, ident!(test));
+    expected
+        .trace_columns
+        .push(trace_segment!(0, "$main", [(a, 1), (b, 1), (c, 4)]));
+    expected.public_inputs.insert(
+        ident!(inputs),
+        PublicInput::new(SourceSpan::UNKNOWN, ident!(inputs), 2),
+    );
+    expected.integrity_constraints = Some(Span::new(
+        SourceSpan::UNKNOWN,
+        vec![enforce!(eq!(access!(a), int!(0)))],
+    ));
+    expected.boundary_constraints = Some(Span::new(
+        SourceSpan::UNKNOWN,
+        vec![
+            let_!(x = lc!(((i, range!(0..3)), (c, expr!(access!(c)))) => mul!(exp!(int!(2), access!(i)), access!(c))).into() =>
+                  enforce!(eq!(bounded_access!(a, Boundary::First), add!(add!(add!(access!(x[0]), access!(x[1])), access!(x[2])), access!(x[3]))))),
```

Comment:
note: similar issue to the above here - this is invalid, but could be made valid by removing the trace column from the list comprehension. The boundary constraint tests below all seem to have the same issue, but I won't note it again